### DISCUSSION TITLE
How do I `brew install` a formula to specified location?

### DISCUSSION THREAD

**Discussion by tddschn** [Homebrew/discussions#4534] [Everyday usage] [answered]
### Description of issue

I need to test a previous version of a formula (the formula is obtained from git history, and there's no official `formula_name@version` formula for the formula. 

To avoid messing with BREW_PREFIX, I want to install the said formula into another location, how do I do that?

If this isn't officially supported / encourged by homebrew, is there a work around?  

Thank you.

**Comment by ZhongRuoyu** [comment, marked as answer]
It is not supported to install a formula to a customised location. However, `brew extract` ([manpage](https://docs.brew.sh/Manpage#extract---version---force-formula-tap)) should help you extract the old version into your own tap; you can then install the extracted version:

```shell
brew tap homebrew/core  # only needed if you are extracting a formula from homebrew/core
brew tap-new tddschn/$tap_name
brew extract $formula tddschn/$tap_name --version=$version
brew install tddschn/$tap_name/$formula@$version
```

You need to give your tap a name (`$tap_name`). Also, substitute `$formula` with the name of the formula you want to install, and `$version` with its version.

**Comment by carlocab** [comment]
You can also install another `brew` into your chosen location to install the formula you want to there. See https://docs.brew.sh/Installation#untar-anywhere-unsupported

**Reply by tddschn** [reply]
Thanks Carlo for the answer. Does Homebrew have any plans of supporting custom formula installation location? If not, why? 

I think this feature would offer great flexibility to certain advanced users and some devs, and make certain workflows smoother.

**Reply by carlocab** [reply]
No, it's just too niche a use case to be worth investing resources in developing and maintaining. See also: https://docs.brew.sh/Installation#multiple-installations-unsupported
